grails-app/services/org/pih/warehouse/report/NotificationService.groovy

```diff
@@ -229,25 +229,10 @@ class NotificationService {
     }
 
 
-    List<Person> resolveNotificationRecipients(Requisition requisition) {
-        List<Person> recipients = []
-        Requisition.withSession {
-            if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
-                if (requisition.approvers?.size()) {
-                    recipients = Person.findAllByIdInList(requisition.approvers.collect { it.id })
-                } else {
-                    // if there are not assigned users as approvers then notify all approvers
-                    recipients = userService.findUsersByRoleTypes(requisition.origin, [RoleType.ROLE_REQUISITION_APPROVER])
-                }
-            }
-        }
-        return recipients
-    }
-
     void publishRequisitionStatusTransitionNotifications(Requisition requisition) {
         switch(requisition.status) {
             case RequisitionStatus.PENDING_APPROVAL:
-                List<Person> recipients = resolveNotificationRecipients(requisition)
+                List<Person> recipients = Person.getAll(requisition.approvers*.id)
```

Comment:
also be aware of the fact that `getAll` would eventually fill not found recipients with `null` values and you might loop through that somewhere. 
It would be 0,00001% case, but just saying.